### DISCUSSION TITLE
Do we have a callback to identify when fallback to mainThread rendering happens ?

### DISCUSSION THREAD

**Discussion by NitishGadangi** [airbnb/lottie-ios#1893] [Q&A] [answered]
## Which Version of Lottie are you using?

Lottie 4.0.1

## Question 
With new `RenderingEngineOption.automatic` set to default. At a few cases the operation falls back to `mainThread`. Which as suggested when explicitly set with `LottieConfiguration(renderingEngine: .mainThread)` has been reducing some over head on CPU.

But for lotties which are being fetched at runtime from cloud, whats the best practice. Do we have some call back which gives that the given lottie is being fall back to `RenderingEngineOption.mainThread`. So that we can explicitly set engine to `mainThread` at runtime.

**Comment by calda** [comment, marked as answer]
I'd say the best-practice is to rely on `RenderingEngineOption.automatic` to automatically handle displaying the remote animation using the best rendering engine for that specific animation. `.automatic` comes with some initial setup overhead but not enough to worry about imo unless you have a very performance-sensitive use case. Otherwise you would have to do something like store a mapping between asset URLs and the rendering engine you'd like to use when displaying that animation.

**Reply by NitishGadangi** [reply]
Thanks for the response.
